### PR TITLE
Replace std::endl with \n

### PR DESCRIPTION
**Pull request by reuk**
`endl` forces flushing of the current stream, which is generally not necessary. By avoiding `endl`, our software becomes faster and more concise, and specifies the required behaviour more accurately. Also, our coding guideline specifies that we should prefer '\n' to `endl`.

### CODE REVIEW

**Comment by reuk** [comment]
I've implemented the quick fixes. I'll try to look at adapting the linter this evening.

**Comment by reuk** [comment]
Hopefully the linter does the right thing now.

File: scripts/cpplint.py [hunk 1]
```diff
@@ -6198,6 +6199,12 @@ def CheckNamespaceOrUsing(filename, clean_lines, linenum, error):
     error(filename, linenum, 'readability/namespace', 4,
           'Do not use using')
 
+def CheckForEndl(filename, clean_lines, linenum, error):
+  """Check that the line does not contain std::endl."""
+  line = clean_lines.elided[linenum]
+  if Match(r'[^a-zA-Z0-9_]*std::endl[^a-zA-Z0-9_]*', line):
+    error(filename, linenum, '', 4, 'Do not use std::endl')
```
Comment (tautschnig) [inline scripts/cpplint.py]: The `''` should be `'runtime/endl'`

Comment (reuk) [inline scripts/cpplint.py]: Well spotted!

**Comment by tautschnig** [comment]
A rebase is needed to make the tests pass.

**Comment by reuk** [comment]
This builds cleanly now.

**Comment by kroening** [comment]
Looks ready to go once rebased.

**Comment by reuk** [comment]
Rebased now.

**Pull request merged**